site/app/controllers/admin/AdminGradeableController.php

```diff
@@ -143,6 +160,9 @@ public function uploadGradeable() {
             return JsonResponse::getSuccessResponse($_POST['id']);
         }
         catch (\Exception $e) {
+            if ($e->getMessage() === 'Date validation failed') {
+                return JsonResponse::getErrorResponse('Date validation has failed, ensure the dates are in the correct sequential order');
```

Comment:
You could catch this as a ValidationException and then return the actual Date error(s) which exists to provide more info of which Date is wrong